stock_back2back_order_proc/stock.py

```diff
@@ -0,0 +1,552 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (c) 2010-2015 Elico Corp. All Rights Reserved.
+#    Author: Yannick Gouin <yannick.gouin@elico-corp.com>
+#            Alex Duan <alex.duan@elico-corp.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import fields, orm
+from openerp.tools.translate import _
+from openerp import netsvc
+
+
+class StockLocation(orm.Model):
+    _name = 'stock.location'
+    _inherit = "stock.location"
+    _columns = {
+        'retention_mode': fields.selection(
+            [('retention', _('Retention Mode')), ('thru', _('Thru mode'))],
+            _('Retention Mode'),
+            required=True,
+            help=_("In 'Retention mode' the system wait for the\
+                  whole quantity before the stuff is processed.\n" +
+                   "In 'Thru mode' the shipped quantity is processed regardless\
+                  of the ordered quantity.")
+        ),
+    }
+    _defaults = {
+        'retention_mode': 'retention',
+    }
+
+
+class StockPicking(orm.Model):
+    _name = 'stock.picking'
+    _inherit = "stock.picking"
+
+    def get_move_chain(self, cr, uid, move_id, context=None, move_obj=False):
+        '''Recursively get the chained moves
+        @return list of the chained moves
+        '''
+        if not move_obj:
+            move_obj = self.pool.get('stock.move')
+        move_tbc = move_obj.browse(cr, uid, move_id, context, move_obj)
+
+        # If there is move_dest_id in the chain
+        if move_tbc.move_dest_id:
+            move_chain = self.get_move_chain(
+                cr, uid, move_tbc.move_dest_id.id, context)
+        else:
+            move_chain = []
+
+        move_chain.append(move_tbc)
+        return move_chain
```

Comment:
you are returning a list of browse records, not a list of ids, so this method must not be exposed to xmlrpc. You should rename it `_get_move_chain`